flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogStateBackend.java

```diff
@@ -302,4 +354,14 @@ private long getMaterializationID(KeyedStateHandle keyedStateHandle) {
             return 0L;
         }
     }
+
+    private Collection<ChangelogStateBackendHandle> reboundCheckpoint(
+            Collection<ChangelogStateBackendHandle> stateBackendHandles) {
+        return stateBackendHandles.stream()
+                .map(
+                        changelogStateBackendHandle ->
+                                changelogStateBackendHandle.rebound(
+                                        changelogStateBackendHandle.getCheckpointId()))
```

Comment:
I think the reason for this call is not obvious but is important, so it should be clarified here in the comment.

Reply:
Agree. I have added detailed comment.

Reply:
Thanks!